MediaBrowser.Controller/Library/ILibraryManager.cs

```diff
@@ -410,6 +410,13 @@ public interface ILibraryManager
         /// <returns><c>true</c> if [is audio file] [the specified path]; otherwise, <c>false</c>.</returns>
         bool IsAudioFile(string path);
 
+        /// <summary>
+        /// Gets the series name from path.
+        /// </summary>
+        /// <param name="path">The path.</param>
+        /// <returns>ItemInfo.</returns>
+        ItemLookupInfo GetSeriesFromPath(string path);
```

Comment:
I know that we already have `GetSeasonNumberFromPath` in this interface, but I do wonder if it would not be better suited for a new `SeriesResolver`. It's pretty terrible design to have media type specific methods in the library manager imo.

Or maybe using EpisodePathParser directly in tmdb is better?

Or better yet, extending `EpisodeInfo` with the SeriesName so it's available to all providers automatically? `EpisodeInfo` is populated here https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.Controller/Entities/TV/Episode.cs#L305

Reply:
Yeah, I don't really like that the name have to be "cleaned" inside the provider. It should be fed a clean name, but it's not happening for the series provider right now. (it gets things like "foobar s01e1" which makes tmdb fail).

I'll see if I can figure out something better than the clean regex.

Reply:
At a glance I don't see any problems with extending `GetLookupInfo` for Series, Season and Episode to include the clean series name. And shouldn't it fix the issue as well (all metadata providers will have to be updated to use it of course)?